### DISCUSSION TITLE
What's the best way to opt out of Alt?

### DISCUSSION THREAD

**Discussion by mikezackles** [helix-editor/helix#6853] [Q&A]
I'm just getting started playing with this editor and there are a lot of things to like.

Hopefully I'm missing something, but as a new user it's seeming to me like the defaults bind a bunch of things to both Ctrl and Alt. I use i3/sway, which ties up a modifier key, so an arrangement that eats up all my modifier keys doesn't really work for me. What are others doing to work around this? Do I just need to manually remap everything I'm interested in?

Thanks!

**Comment by mikezackles** [comment]
As a starting point, I'd love to just move these into a submenu of the space menu using the same mappings, but I'm not sure if this is possible. I'm fine with doing this manually, but I'm not sure how to do so. Is there a way to add things to the space menu in the config file?

**Comment by ysooqe** [comment]
I would like to second this question in general. But unfortunately, I do not think there is a general or an "intended" solution for this (yet?).

My "solution(s)": At first, I had printed out my WM modifier mappings (using `cat` and `grep`), looked through the official keymap documentation of Helix and remapped everything that was already taken.
Since I kind of do not want to remap anything though, I have recently switched my WM config to use the `Super` (Windows) key instead of the `Alt` key as the primary modifier. This way, I do not need to remap anything in Helix and all the muscle memory I need to "relearn" for the WM is moving my left thumb a single key to the left every time. It's not optimal but it's convenient enough.

**Comment by djpowers** [comment]
I'm also looking for an alternative to the `Alt-*` keymaps.

You should be able to add items to space mode with something like the following (see #2844):
```toml
[keys.normal.space]
u = "earlier"
```
Be aware you're more likely to run into key collisions since space mode is already pretty well-populated.

You could also create a new minor mode to house the alt-based commands:
```toml
[keys.normal.D]
s = "whatever alt-s would be normally"
```
(Credit to [this Reddit post](https://www.reddit.com/r/HelixEditor/comments/1730bm5/anyone_hates_the_alt_key/k407ffw/) for the technique of adding a minor mode)

Out of curiosity, I mapped out the [normal mode](https://docs.helix-editor.com/keymap.html#normal-mode) alt-keymaps by category:
## Alt-based keymaps
### Movement
- "." = "repeat_last_motion"
### Changes
- "`" = "switch_to_uppercase"
- "u" = "earlier"
- "U" = "later"
- "d" = "delete_selection_noyank"
- "c" = "change_selection_noyank"
### Shell
- "|" = "shell_pipe_to"
- "!" = "shell_append_output"
### Selection manipulation
- "s" = "split_selection_on_newline"
- "minus" = "merge_selections"
- "_" = "merge_consecutive_selections"
- ";" = "flip_selections"
- ":" = "ensure_selections_forward"
- "," = "remove_primary_selection"
- "C" = "copy_selection_on_prev_line"
- "(" = "rotate_selection_contents_backward"
- ")" = "rotate_selection_contents_forward"
- "x" = "shrink_to_line_bounds"
- "J" = "join_selections_space"
- "K" = "remove_selections"
- "o" = "expand_selection"
- "up" = "expand_selection"
- "i" = "shrink_selection"
- "down" = "shrink_selection"
- "p" = "select_prev_sibling"
- "left" = "select_prev_sibling"
- "n" = "select_next_sibling"
- "right" = "select_next_sibling"

There was another discussion opened in #8731 about adding another minor mode to house the selection commands, which largely make use of the alt key. That would free up a lot of keys, but you could remap this manually for now.

**Reply by sicher** [reply]
The [keymap brainstorm](https://github.com/helix-editor/helix/wiki/Keymap-brainstorm) wiki page explicitly says ”we don’t want to use alt”. Is that still a goal? I personally would like to try getting rid of all alt-keymaps, if possible.

**Reply by sicher** [reply]
I'm experimenting with putting a bunch of selection manipulation maps under the "m" minor mode. The idea is to treat "m" as "manipulate selection" (or "modify selection") instead of "match".

In my mind, everything currently under "m" apart from "match_brackets" would fit under "manipulate selection". The "match_brackets" binding can probably be moved to "g" (goto minor mode).
